src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -334,5 +334,37 @@ public static bool IsMountDisabled(this IEnvironment environment)
             return string.Equals(environment.GetEnvironmentVariable(MountEnabled), "0")
                 || string.IsNullOrEmpty(environment.GetEnvironmentVariable(MeshInitURI));
         }
+
+        public static CloudName GetCloudName(this IEnvironment environment)
+        {
+            var cloudName = environment.GetEnvironmentVariable(EnvironmentSettingNames.CloudName);
+            if (Enum.TryParse(cloudName, true, out CloudName cloud))
+            {
+                return cloud;
+            }
+
+            return CloudName.Azure;
+        }
+
+        public static string GetStorageSuffix(this IEnvironment environment)
+        {
+            switch (GetCloudName(environment))
+            {
+                case CloudName.Azure:
+                    return CloudConstants.AzureStorageSuffix;
+                case CloudName.Blackforest:
+                    return CloudConstants.BlackforestStorageSuffix;
+                case CloudName.Fairfax:
+                    return CloudConstants.FairfaxStorageSuffix;
+                case CloudName.Mooncake:
+                    return CloudConstants.MooncakeStorageSuffix;
+                case CloudName.USNat:
+                    return CloudConstants.USNatStorageSuffix;
+                case CloudName.USSec:
+                    return CloudConstants.USSecStorageSuffix;
+                default:
```

Comment:
Is this default to ensure we're handling out of Azure scenarios? If so, it would be good to have better logic here to handle that as assuming this enumeration will never change is a potential future bug.

Reply:
The default value here is just making the current assumption explicit. Not sure I understand what the potential bug is. As long as the correct cloud is passed as env variable, this shouldn't break. We can also assume these enum values will not change (new ones might be added). Let me know if I misunderstood your question..

Reply:
Can you expand on what you mean by "making the current assumption explicit"? 

About the comments: 

>  As long as the correct cloud is passed as env variable, this shouldn't break

That's part of the problem, this is just assuming that will be the case and if it ever isn't (e.g. a new environment or a defect), the behavior will be incorrect.

> We can also assume these enum values will not change (new ones might be added).

We can't guarantee that, and handling this scenario in the logic would help us prevent a bug from making its way into prod, or make this significantly easier to identify the problem if it ever does.

> Let me know if I misunderstood your question..

Nope! I think you got it :)